pkg/handler/handler.go

```diff
@@ -126,12 +130,19 @@ func New(options Options) (sdk.Handler, error) {
 	if options.Handle != nil {
 		handle = options.Handle
 	}
-	return &handler{crdToPlaybook: options.GVKToRunner, handle: handle}, nil
+	return &handler{
+		crdToPlaybook: options.GVKToRunner,
+		handle:        handle,
+		eventHandlers: []events.EventHandler{
+			events.NewLoggingEventHandler(events.Tasks),
+		},
+	}, nil
 }
 
 type handler struct {
 	crdToPlaybook map[schema.GroupVersionKind]runner.Runner
 	handle        EventHandler
+	eventHandlers []events.EventHandler
```

Comment:
I follow what's going on here, but I find it confusing to keep track of the many kinds of event handlers there are. I wonder if there's a way for us to make it less confusing for people who will try to use it.

This handler struct is an event handler, it has its own event handler (another event handler struct), plus now has these new event handlers. There's the SDK's idea of event handler, this package's definition of an event handler, and the ansible-runner event handler all colliding.

Maybe it would help to lay out a more complete vision for how a developer would use this as a library in the hybrid case. Here are the first questions that come to mind: This type isn't exported, so are we assuming they'll just modify this code to their purposes? Do we expect them to use the `defaultHandle` function? Or provide their own? Or modify it to their purposes?

At least as the code sits in this PR, I don't see that it's easier to add event handling behavior as part of this struct as opposed to just doing whatever I want in the loop over `eventChan`. I suspect you have a future vision in mind, and it would be helpful to get more insight.

Reply:
Eventually, we would add this to the options and that is how a user would add or remove them. 

I think we should consider if we want make this a default that always gets added or not. 

Re: concern about overloading of terms. If we want to re-think the names I am not opposed, but I think this change is a step in the right direction and should not be held due to naming concerns. Rember that the sdk.Events and Handler will be changing the future to Reconciler and ReconcileRequests.